Backend/django_mobx_test/restaurant/views.py

```diff
@@ -0,0 +1,182 @@
+            user = serializer.validated_data['user']
+            print(user.id)
+            print("after verification")
+
+            token, created = Token.objects.get_or_create(user=user)
+            user_serializer = UserSerializer(user)
+            return Response({'token': token.key, 'user': user_serializer.data})
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+class RestaurantViewSet(ViewSet):
+    permission_classes = [IsAuthenticated, ]
+    serializer_class = RestaurantSerializer
+
+    def create(self, request):
+        try:
+            serializer = RestaurantSerializer(data=request.data)
+            serializer.is_valid(raise_exception=True)
+            restaurant = Restaurant(name=serializer.validated_data['name'], owner=request.user)
+            restaurant.save()
+            restaurant_data = RestaurantModelSerializer(restaurant)
+            return Response({'data': restaurant_data.data}, status=status.HTTP_201_CREATED)
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    def list(self, request):
+        try:
+            restaurant = Restaurant.objects.filter(owner=request.user)
+            restaurant_data = RestaurantModelSerializer(restaurant, many=True)
+            return Response({'data': restaurant_data.data}, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    def retrieve(self, request, pk=None):
+        try:
+            restaurant = Restaurant.objects.get(id=pk, owner=request.user)
+            restaurant_data = RestaurantModelSerializer(restaurant)
+            return Response({'data': restaurant_data.data}, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    def destroy(self, request, pk=None):
+        try:
+            restaurant = Restaurant.objects.get(id=pk, owner=request.user)
+            tickets = Ticket.objects.filter(restaurant=restaurant)
+            for ticket in tickets:
+                ticket.delete()
+
+            restaurant.delete()
+            return Response({'data': status.HTTP_200_OK}, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    # Tickets
+    @action(detail=True, methods=['post'])
+    def ticket(self, request, pk=None):
+        try:
+            serializer = TicketSerializer(data=request.data)
+            serializer.is_valid(raise_exception=True)
+            restaurant = Restaurant.objects.get(id=pk,  owner=request.user)
+            ticket = Ticket(name=serializer.validated_data['name'], amount=serializer.validated_data['amount'],
```

Comment:
In general workflow of DRF, the record should be stored in serializer like create or update method, which is good experience. Not sure why you use serializer for only validation and use model save on view